packages/material-ui/src/Select/SelectInput.js

```diff
@@ -141,6 +150,22 @@ const SelectInput = React.forwardRef(function SelectInput(props, ref) {
     event.preventDefault();
     displayRef.current.focus();
 
+    const doc = ownerDocument(event.target);
+    if (handleClick.current) {
+      doc.removeEventListener('click', handleClick.current, true);
+    }
+
+    // Since the targets of the `mousedown` and `mouseup` events are different,
+    // the browser will dispatch the `click` to a common ancestor.
+    // This stops the next `click` and re-dispatch it with the right target.
+    handleClick.current = (clickEvent) => {
+      clickEvent.stopPropagation();
+      doc.removeEventListener('click', handleClick.current, true);
+      event.target.dispatchEvent(new window.Event('click', { bubbles: true }));
```

Comment:
This shows how error prone the solution is. 
```suggestion
      event.target.dispatchEvent(new doc.defaultView.MouseEvent('click', { bubbles: true, cancelable: true }));
```